torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -796,25 +797,29 @@ void ProcessGroupNCCL::WorkNCCL::abort() {
   ncclCommDevIdxMapMutex.unlock();
 }
 
-ProcessGroupNCCL::CUDAEventCache::CUDAEventCache() = default;
+ProcessGroupNCCL::CUDAEventCache::CUDAEventCache()
+    : caches_(at::cuda::device_count()) {}
```

Comment:
If we use map, we wouldn't need to allocate many up front?

Reply:
Well, when you have many devices, you are going to allocate anyway.. 

Also computing hash is more expensive than index look-up, lol

Reply:
I disagree, here we are allocating `device_count()` many, whether they are used or not. 
Whereas, in most cases, we only use 1, maybe in corner cases 2.

Reply:
What is the value of `device_count()`? Isn't it always 1?

Reply:
And after we retire DDP's multi-device support, we will have to rework this code if it is as-is.